src/qt/coincontroldialog.cpp

```diff
@@ -634,6 +634,12 @@ void CoinControlDialog::updateLabels()
             nBytesInputs++;
     }
 
+    // if all coins are unchecked, reset SelectAll (if previously set to UnselectAll)
+    // this happens  when the user manually unchecks all entries (after SelectAll)
+    // or after a spend, when coinControl is cleared.
+    if (coinControl->QuantitySelected() == 0 && ui->pushButtonSelectAll->isChecked())
```

Comment:
This may be a little more user friendly if QuantitySelected is less then the QuantitySelected when they first clicked the select all.  e.g. if they deselect one, it goes back to "select all".  But then again, maybe they might want to decide to deselect all.  (random thoughts through my fingertips), maybe "quantitySelected is less than half the original quantityselected"... e.g. when there's more selected then unselected; it gives you 'deselect all", when there's more unselected then selected, it switches to 'select all'.  Might need to read my Human Factors book again.

Reply:
agreed. we can do that.